Orange/widgets/unsupervised/owdistances.py

```diff
@@ -160,6 +161,8 @@ def checks(metric, data):
         except MemoryError:
             self.Error.distances_memory_error()
             return
+
+        met = np.nan_to_num(met)
```

Comment:
@janezd, **Distance Transformation** widget could have a check box where user could select to impute infinite distances. @lanzagar suggested something like that.